gapic/schema/api.py

```diff
@@ -526,24 +532,73 @@ def _get_methods(self,
                         'metadata type.',
                     )
                 lro = wrappers.OperationInfo(
-                    response_type=self.api_messages[address.resolve(
+                    response_type=self.api_messages[service_address.resolve(
                         op.response_type,
                     )],
-                    metadata_type=self.api_messages[address.resolve(
+                    metadata_type=self.api_messages[service_address.resolve(
                         op.metadata_type,
                     )],
                 )
 
+            # If we got a gRPC service config, get the appropriate retry
+            # and timeout information from it.
+            retry = None
+            timeout = None
+            if self.opts.retry:
+                # This object should be a dictionary that conforms to the
+                # gRPC service config proto:
+                #   Repo: https://github.com/grpc/grpc-proto/
+                #   Filename: grpc/service_config/service_config.proto
+                #
+                # We only care about a small piece, so we are just leaving
+                # it as a dictionary and parsing accordingly.
+                for mc in self.opts.retry.get('methodConfig', []):
```

Comment:
This confused me a bit at first; the structure of the loop implies that there might be more than one match.

This is a little more clear in terms of data and control flow IMO, plus the comment:
```python
# Search for the method config for this method, if one exists.
mc = next((mc
           for mc in self.opts.retry.get('methodConfig', [])
           if selector in mc.get('name')), None)
if mc:
   ...
```

Reply:
Done, but this changed coverage slightly, so now the test has three methods instead of two. This is better anyway.